### DISCUSSION TITLE
Ability to skip request/response interceptors on per request basis

### DISCUSSION THREAD

**Discussion by amitsainii** [axios/axios#5075] [Ideas]
With Axios v1.x we do get this awesome `runWhen` option to run request interceptors conditionally. I'm proposing the following 2 ideas that I can easily contribute myself 🤞 

- Add support for `runWhen` in the response interceptors as well.
- Adding 2 booleans `skipRequestInterceptors` and `skipResponseInterceptors` that defaults to `false` to `AxiosRequestConfig`. I know we can easily skip request interceptors on a per-request basis by simply setting `runWhen: () => false`, but this isn't very explicit for a scenario where we simply want to skip interceptors. Aso this only supports request interceptors for now. Having 2 additional booleans can be pretty useful.

I'd love to contribute this feature if everyone's on board with the ideas :)

**Comment by amitsainii** [comment]
@jasonsaayman Do you have any thoughts on this?

**Reply by amitsainii** [reply]
@DigitalBrainJS @jasonsaayman  following up to see if you think this feature can be of a value

**Comment by DigitalBrainJS** [comment]
I'm not sure that at the moment it is necessary to complicate the implementation. If I have enough energy, time and enthusiasm, in the future Axios will get the functionality of hooks, which are much more flexible and powerful and probably cover this functionality. It is currently in the early stages of R&D.

```js
axios.get('http://localhost/', {
  hooks: {
    'request:prepend': async ({config}, next) => {
      if (config.url === '/foo') { // skip all following hooks for paths other than /foo
        return await next(); // call the next hooks in the stack
      }

      if (config.url === '/bar') { // early resolution
        return AxiosResponse.from({
          status: 200, // optionally
          data: "myData"
        });
      }
    }
  }
})
```

**Comment by ErTy-Dev** [comment]
I think you mean this one  
` axios.interceptors.request.clear();`
